include/boost/regex/v4/fileiter.hpp

```diff
@@ -229,14 +229,15 @@ class BOOST_REGEX_DECL mapfile_iterator
       file = f;
       node = f->_first + arg_position / mapfile::buf_size;
       offset = arg_position % mapfile::buf_size;
-      file->lock(node);
+      if(file && node)
```

Comment:
Hi John,

I most certainly defer to you on these issues :+1: !  Do you need me to make any changes or can you make them?  if the former, what would you like me to do?

Regards,

Moshe